jax/core.py

```diff
@@ -1802,7 +1804,7 @@ def subst_axis_names_jaxpr(jaxpr: Union[Jaxpr, ClosedJaxpr], subst: AxisSubst):
   if isinstance(jaxpr, ClosedJaxpr):
     consts = jaxpr.consts
     jaxpr = jaxpr.jaxpr
-  var_map: Dict[Var, Var] = {}
+  var_map: Dict[Var, Var] = {unitvar: unitvar}
```

Comment:
Why this change?

Reply:
This PR enables calling `subst_axis_names_jaxpr` on control flow primitive JAXprs, which apparently have unitvars in them. This function previously could not handle unitvars even though the substitution is trivial.

Reply:
I'm pretty sure it did handle unit vars. Check out the `if` in `subst_axis_names_var`. I'd rather avoid adding this here if possible.

Reply:
It handles `unitvars` in the inputs but not the outputs. This line
```
outvars: List[Atom] = [v if isinstance(v, Literal) else var_map[v] for v in jaxpr.outvars]
```
fails if there are any unitvars in `jaxpr.outvars`. I wasn't sure of a simpler solution than this.

Reply:
Ohh I see, good point. Is that line necessary for the tests for this PR to pass? If no then let's submit it as a separate change.

Reply:
It is necessary unfortunately.